### PR TITLE
MULE-19038: feature flag reserved variable names

### PR DESCRIPTION
**Pull request by szaffarano**

### CODE REVIEW

File: src/main/java/org/mule/runtime/api/util/MuleSystemProperties.java [hunk 1]
```diff
@@ -259,6 +260,15 @@
...
   public static final String MULE_LAX_ERROR_TYPES = SYSTEM_PROPERTY_PREFIX + "errorTypes.lax";
 
+  /**
+   * When set to {@code true} an application won't be able to override reserved properties such as <code>app.name</code>. If it
+   * isn't set, the behaviour will depend on {@link MuleRuntimeFeature#HONOUR_RESERVED_PROPERTIES}.
+   * 
+   * @see MuleRuntimeFeature#HONOUR_RESERVED_PROPERTIES
+   * @since 4.4.0, 4.3.1
+   */
+  public static final String HONOUR_RESERVED_PROPERTIES_PROPERTY = SYSTEM_PROPERTY_PREFIX + "honour.reserved.properties";
```
Comment (elrodro83) [inline src/main/java/org/mule/runtime/api/util/MuleSystemProperties.java:270]: overrides should be managed by the feature service internally, in a generic standard way.

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: src/main/java/org/mule/runtime/api/util/MuleSystemProperties.java (hunk 1 above)
Comment (szaffarano) [reply src/main/java/org/mule/runtime/api/util/MuleSystemProperties.java:270]: I left the System Property constant definition here, but the logic that uses it remains internally to the Feature Flagging mechanism.  Do you think that we should define this constant only [here](https://github.com/mulesoft/mule-api/pull/748/files#diff-9018e4b400b25f6397972e30bf9832135f6557d83869174a7c8a99e87f20f32aR64)?

File: src/main/java/org/mule/runtime/api/config/MuleRuntimeFeature.java [hunk 2]
```diff
@@ -38,7 +41,8 @@
  *       * @since 4.4.0, 4.3.1
  *       *
  *      {@code *}/
- *      HONOUR_RESERVED_PROPERTIES("Whether reserved properties such as app.name can't be overridden by global properties.", "MULE-17659", "4.4.0, 4.3.1"),
+ *      HONOUR_RESERVED_PROPERTIES("Whether reserved properties such as app.name can't be overridden by global properties.",
```
Comment (elrodro83) [inline src/main/java/org/mule/runtime/api/config/MuleRuntimeFeature.java:44]: `app.name` -> `'app.name'`

File: src/main/java/org/mule/runtime/api/config/MuleRuntimeFeature.java [hunk 3]
```diff
@@ -48,20 +52,29 @@
...
-  ;
+
+  /**
+   * An application shouldn't override reserved properties, but it was possible until MULE-17659. This behaviour has to be fixed
+   * by default to any application developed for 4.3.0+ Runtime but can be overridden by setting
+   * {@link MuleSystemProperties#HONOUR_RESERVED_PROPERTIES_PROPERTY} System Property.
+   * 
+   * @since 4.4.0, 4.3.1
+   */
+  HONOUR_RESERVED_PROPERTIES("Whether reserved properties such as app.name can't be overridden by global properties.",
```
Comment (elrodro83) [inline src/main/java/org/mule/runtime/api/config/MuleRuntimeFeature.java:63]: `app.name` -> `'app.name'`

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by elrodro83 — DISMISSED** [review]

File: src/main/java/org/mule/runtime/api/util/MuleSystemProperties.java (hunk 1 above)
Comment (elrodro83) [reply src/main/java/org/mule/runtime/api/util/MuleSystemProperties.java:280]: nah, it's okay to have it here as long as the doc references the proper flag it modifies.

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**